ui/src/auth/Purgatory.js

```diff
@@ -1,70 +1,164 @@
-          </h3>
-          {role === MEMBER_ROLE ? memberCopy : viewerCopy}
-          <p>Contact your Administrator for assistance.</p>
-          <hr />
-          <pre>
-            <code>
-              username: {name}
-              <br />
-              provider: {provider}
-              <br />
-              scheme: {scheme}
-            </code>
-          </pre>
+    const rolesAndOrgs = organizations.map(organization => ({
+      organization,
+      role: getRoleNameByOrgID(organization.id, roles),
+      currentOrganization: organization.id === currentOrganization.id,
+    }))
+
+    const subHeading =
+      rolesAndOrgs.length === 1
+        ? 'Authenticated in 1 Organization'
+        : `Authenticated in ${rolesAndOrgs.length} Organizations`
+
+    return (
+      <div>
+        <div className="auth-page">
+          <div className="auth-box">
+            <div className="auth-logo" />
+            <div className="auth--purgatory">
+              <h3>
+                {name}
+              </h3>
+              <h6>
+                {subHeading}{' '}
+                <code>
+                  {scheme}/{provider}
+                </code>
+              </h6>
+              {rolesAndOrgs.length
+                ? <div className="auth--list">
+                    {rolesAndOrgs.map(rag =>
+                      <div
+                        key={rag.organization.id}
+                        className={
+                          rag.currentOrganization
+                            ? 'auth--list-item current'
+                            : 'auth--list-item'
+                        }
+                      >
+                        <div className="auth--list-info">
+                          <div className="auth--list-org">
+                            {rag.organization.name}
+                          </div>
+                          <div className="auth--list-role">
+                            {rag.role}
+                          </div>
+                        </div>
+                        {rag.role === MEMBER_ROLE
+                          ? <span className="auth--list-blocked">
+                              Contact your Admin<br />for access
+                            </span>
+                          : <button
+                              className="btn btn-sm btn-primary"
+                              onClick={this.handleClickLogin(rag.organization)}
+                            >
+                              Login
+                            </button>}
+                      </div>
+                    )}
+                  </div>
+                : <p>You are a Lost Soul</p>}
+              <a href={logoutLink} className="btn btn-sm btn-link auth--logout">
+                Logout
+              </a>
+            </div>
+          </div>
+          <p className="auth-credits">
+            Made by <span className="icon cubo-uniform" />InfluxData
+          </p>
+          <div className="auth-image" />
```

Comment:
All of this markup looks solid.  It would be much easier to read and more declarative if we broke this out into a `<Header>` and something like  a `<RoleOrg>`

Reply:
Done